test/remote/gateways/remote_datatrans_test.rb

```diff
@@ -220,4 +235,17 @@ def test_successful_void_with_google_pay
     response = @gateway.void(authorize_response.authorization, @options)
     assert_success response
   end
+
+  def test_successful_purchase_with_3ds
+    response = @gateway.purchase(@amount, @credit_card_frictionless, @options.merge(@three_d_secure))
+    assert_success response
+  end
+
+  def test_failed_purchase_with_3ds
```

Comment:
💬 Comment:

Mmmm this test is not testing a "failed purchase with 3DS" it just testing if the request structure make sense by not sending the proper format,  would be great if we have something that actually causes a decline on the gateway side.

Reply:
I cannot find something that makes this fail (specifically for 3DS).